src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -306,8 +327,13 @@ ApiGlobe.prototype.setRealisticLightingOn = function(value) {
 ApiGlobe.prototype.setLayerVisibility = function(id, visible) {
 
     this.scene.getMap().setLayerVisibility(id, visible);
-
+    eventLayerChanged.layerId = id;
```

Comment:
Need Factorization 

Listen all Layers changes (visibilty, opacity, id...) and `dispatchEvent(eventLayerChanged)`